js/ui/appDisplay.js

```diff
@@ -1731,6 +1735,10 @@ const AppIcon = new Lang.Class({
 
     popupMenu: function() {
         this._removeMenuTimeout();
+
+        if (!this._showMenu)
+            return true;
```

Comment:
That's exactly my reasoning for returning _true_. Also I kind of figured it would be used to propagate an event which is why code like this usually returns a boolean in GTK/Glib...
I also noticed there are no users of the return code but this is Js and I don't know how safe it is to just remove the return without thoroughly testing every thing.